src/main/webapp/resources/js/components/main-navigation/components/AnnouncementsModal.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect, useState } from "react";
+import { Button, Space, Tag, Typography } from "antd";
+import { formatDate } from "../../../utilities/date-utilities";
+import {
+  useGetUnreadHighPriorityAnnouncementsQuery,
+  useMarkAnnouncementAsReadMutation,
+} from "../../../redux/endpoints/announcements";
+import ReactMarkdown from "react-markdown";
+import { FlagTwoTone } from "@ant-design/icons";
+import { ScrollableModal } from "../../ant.design/ScrollableModal";
+
+const { Text } = Typography;
+
+/**
+ * React component to render a modal if high priority announcements are available
+ * @constructor
+ */
+export default function AnnouncementsModal(): JSX.Element | null {
+  const { data: announcements, isSuccess } =
+    useGetUnreadHighPriorityAnnouncementsQuery(undefined, {});
+  const [markAnnouncementAsRead] = useMarkAnnouncementAsReadMutation();
+
+  const [visible, setVisible] = useState<boolean>(false);
+  const [index, setIndex] = useState<number>(0);
+
+  useEffect(() => {
+    if (isSuccess) {
+      setVisible(true);
+    }
+  }, [isSuccess]);
+
+  if (!isSuccess || announcements === undefined || announcements.length === 0) {
```

Comment:
Should this not be in the return below? Seems out of place here

Reply:
No it is good here as it prevent any of the code from executing below if there is no need to render anything